Storage/StorageManager.php

```diff
@@ -161,68 +118,177 @@ public function storeUploadedFile($type, UploadedFile $uploadedFile, $newFilenam
      */
     public function storeFileFromLocalPath($type, $path, $newFilename = null, $unlinkAfterStore = false)
     {
-        if (!file_exists($path)) {
-            throw new \InvalidArgumentException(sprintf("Cannot store file from non-existent path at '%s'", $path));
+        $options = [];
+        if ($newFilename !== null) {
+            $options['newFilename'] = $newFilename;
         }
-        $this->validateFileStorageType($type);
-
-        $storage = $this->getStorageFromWritePolicy($type);
+        if ($unlinkAfterStore !== false) {
+            $options['unlinkAfterStore'] = $unlinkAfterStore;
+        }
+        return $this->store($type, $path, $options);
+    }
 
-        if (null === $newFilename) {
-            $newFilename = uniqid('gen' . substr(hash('sha256', $path), 0, 4));
+    /**
+     * Takes a content and store it in permanent storage
+     * Content might be:
+     * UploadedFile
+     * string - path to local file
```

Comment:
I think the string should be a path to a local file, and if _not_ a local file... the content directly.

Reply:
what if this a content from a user that accidentally point to a real file on a server?

Reply:
Thats a good point.  Can you look at some other implementations?  Gaufrette maybe?  The AWS S3 sdk?  Maybe a dropbox SDK?

I seem to recall implementations where $content was either a file or a string directly.

Reply:
Amazon S3 S3Client::puObject - can use 'Body' parameter as file contents, or 'SourceFile' parameter for filepath.
GoogleCloudStorage Storage/Connection/Rest::inserObject - internaly uses GuzzleHttp\Psr7\stream_for($data), so in case of string it would be always file contents.
AzureStorage FileRestProxy::putFileRangeAsync - also uses GuzzleHttp\Psr7\stream_for($content) internaly.

DropBox has separate methods for resource and string content:
Client::uploadFile - uses resource as content
Client::uploadFileAsString - uses string as file contents

no method to upload from filepath

Personally, I don't see how we can distinguish content from filepath without introducing new option.